[16.0][MIG] base_bank_from_iban: Migration to 16.0

pegonzalezspesol opened the pull request.

carlosdauden committed:
[ADD] base_bank_from_iban: Get bank from IBAN number (#32)

pedrobaeza committed:
[MIG] base_bank_from_iban: Migration to 12.0

emagdalenaC2i committed:
[IMP] base_bank_from_iban: black, isort

emagdalenaC2i committed:
[MIG] base_bank_from_iban: Migration to 13.0

OCA-git-bot committed:
pre-commit update

ozono committed:
[MIG] base_bank_from_iban: Migration to 14.0

bosd committed:
Added translation using Weblate (Dutch)

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 100.0% (7 of 7 strings)

Translation: community-data-files-14.0/community-data-files-14.0-base_bank_from_iban
Translate-URL: https://translation.odoo-community.org/projects/community-data-files-14-0/community-data-files-14-0-base_bank_from_iban/nl/

manuelregidor committed:
[MIG]base_bank_from_iban: Migration to 15.0

pedrobaeza committed:
[IMP] base_bank_from_iban: Promote to mature

It complies with the requirements and other modules depending on it are also mature.

pegonzalezspesol committed:
[IMP] base_bank_from_iban: black, isort, prettier

pegonzalezspesol committed:
[MIG] base_bank_from_iban: Migration to 16.0

pedrobaeza commented:
/ocabot migration base_bank_from_iban

pedrobaeza commented on base_bank_from_iban/README.rst:
```diff
@@ -0,0 +1,98 @@
+
+This module adds a code to bank definition for using it as matching for filling the bank
+from the IBAN bank account number. It uses the existing by country bank mapping in
+standard Odoo (https://github.com/odoo/odoo/blob/12b15cb55414d1f6dfc6b3b4e0c38638551ee54d/addons/base_iban/models/res_partner_bank.py#L81-L149),
```
Please update it to https://github.com/odoo/odoo/blob/f5ffcf7feec5526a483f8ddd240648c084351008/addons/base_iban/models/res_partner_bank.py#L105-L175

pedrobaeza commented on base_bank_from_iban/__manifest__.py:
```diff
@@ -0,0 +1,16 @@
+{
+    "name": "Bank from IBAN",
+    "version": "16.0.1.0.0",
+    "author": "Tecnativa, " "Odoo Community Association (OCA)",
```
```suggestion
    "author": "Tecnativa, Odoo Community Association (OCA)",
```

pedrobaeza commented on base_bank_from_iban/models/account_journal.py:
```diff
@@ -0,0 +1,23 @@
+    _inherit = "account.journal"
+
+    @api.onchange("bank_acc_number")
+    def _onchange_bank_acc_number_base_bank_from_iban(self):
```
This one doesn't exists anymore. Remove the entire file.

pedrobaeza commented on base_bank_from_iban/models/res_partner_bank.py:
```diff
@@ -0,0 +1,30 @@
+    _inherit = "res.partner.bank"
+
+    @api.onchange("acc_number", "acc_type")
+    def _onchange_acc_number_base_bank_from_iban(self):
```
Instead of the deprecated onchange, please use computed writable field on `bank_id` with the following convention:

https://github.com/OCA/odoo-community.org/pull/50/files

pedrobaeza replied:
And please check if with code creation, the field is computed as well.

pedrobaeza commented on base_bank_from_iban/tests/test_base_bank_from_iban.py:
```diff
@@ -0,0 +1,56 @@
+    def test_onchange_acc_number_iban(self):
+        partner_bank = self.bank_obj.new()
+        partner_bank.acc_number = "es1299999999509999999999"
+        partner_bank._onchange_acc_number_base_bank_from_iban()
```
Use `Form` instead of manually calling the method.

pedrobaeza committed:
[MIG] base_bank_from_iban: Finish migration

- Regenerate README
- Journal code removed, as you cannot specify the bank account number
  from it.
- Added the equivalente code for the add bank account wizard.

TT40005

pedrobaeza reviewed (approved):
I have fixed the remaining things my self, so good to be merged for continuing with the dependencies:

/ocabot merge nobump
